// buefy/dist/cjs/tag.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Tag = require('./Tag-DfJkWx-r.js');
var vue = require('vue');
var plugins = require('./plugins-AnZKsRZe.js');
require('./Icon-9JyUGVH3.js');
require('./config-O5NiT3Jw.js');
require('./helpers.js');

var _sfc_main = vue.defineComponent({
  name: "BTaglist",
  props: {
    attached: Boolean
  }
});

function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createElementBlock(
    "div",
    {
      class: vue.normalizeClass(["tags", { "has-addons": _ctx.attached }])
    },
    [
      vue.renderSlot(_ctx.$slots, "default")
    ],
    2
    /* CLASS */
  );
}
var Taglist = /* @__PURE__ */ plugins._export_sfc(_sfc_main, [["render", _sfc_render], ["__file", "/home/runner/work/buefy-next/buefy-next/packages/buefy-next/src/components/tag/Taglist.vue"]]);

const Plugin = {
  install(Vue) {
    plugins.registerComponent(Vue, Tag.BTag);
    plugins.registerComponent(Vue, Taglist);
  }
};

exports.BTag = Tag.BTag;
exports.BTaglist = Taglist;
exports.default = Plugin;
