// buefy/dist/esm/tag.js
import { B as BTag } from './Tag-ATQChtTK.js';
import { defineComponent, openBlock, createElementBlock, normalizeClass, renderSlot } from 'vue';
import { _ as _export_sfc, a as registerComponent } from './plugins-y-E4hvZJ.js';
import './Icon-_DiyoPxa.js';
import './config-3Md_fpMO.js';
import './helpers.js';

var _sfc_main = defineComponent({
  name: "BTaglist",
  props: {
    attached: Boolean
  }
});

function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass(["tags", { "has-addons": _ctx.attached }])
    },
    [
      renderSlot(_ctx.$slots, "default")
    ],
    2
    /* CLASS */
  );
}
var Taglist = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["__file", "/home/runner/work/buefy-next/buefy-next/packages/buefy-next/src/components/tag/Taglist.vue"]]);

const Plugin = {
  install(Vue) {
    registerComponent(Vue, BTag);
    registerComponent(Vue, Taglist);
  }
};

export { BTag, Taglist as BTaglist, Plugin as default };
